src/app/services/actions/snippet-action-creator.ts

```diff
@@ -86,3 +87,13 @@ export function getSnippet(language: string): Function {
     }
   };
 }
+
+export function constructHeaderString(sampleHeaders: Header[]): string {
+  let headers = '';
+  if (sampleHeaders && sampleHeaders.length > 0) {
+    sampleHeaders.forEach(header => {
+      headers += `${header.name}: ${header.value}\r\n`;
+    });
+  }
```

Comment:
We should add 'Content-Type: application/json' header if the content-type header doesn't already exist

Reply:
Okay. Let me add that check